test/e2e/functional/tests/e2e.go

```diff
@@ -232,4 +234,102 @@ var _ = Describe("metallb", func() {
+				Expect(err).ToNot(HaveOccurred())
+				controller, err := testclient.Client.Deployments(metallb.Namespace).Get(context.Background(), consts.MetalLBDeploymentName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(len(pods.Items)).To(Equal(int(controller.Status.Replicas)))
+				var controllerContainerFound bool
+				for _, pod := range pods.Items {
+					Expect(pod.Status.Phase).To(Equal(corev1.PodRunning))
+					for _, container := range pod.Spec.Containers {
+						if container.Name == "controller" {
+							Expect(container.Resources).NotTo(BeNil())
+							Expect(container.Resources.Limits.Cpu().MilliValue()).To(Equal(int64(100)))
+							controllerContainerFound = true
+						}
+					}
+				}
+				Expect(controllerContainerFound).To((Equal(true)))
+				Expect(controller.Spec.Template.Spec.PriorityClassName).To(Equal(priorityClassName))
+				Expect(controller.Spec.Template.Annotations["test"]).To(Equal("e2e"))
+				Expect(controller.Spec.Template.Spec.Affinity.PodAffinity.RequiredDuringSchedulingIgnoredDuringExecution[0].LabelSelector.MatchLabels["app"]).To(Equal("metallb"))
+				Expect(controller.Spec.Template.Spec.Affinity.PodAffinity.RequiredDuringSchedulingIgnoredDuringExecution[0].TopologyKey).To(Equal("kubernetes.io/hostname"))
+
+				Eventually(func() bool {
+					daemonset, err := testclient.Client.DaemonSets(metallb.Namespace).Get(context.Background(), consts.MetalLBDaemonsetName, metav1.GetOptions{})
+					if err != nil {
+						return false
+					}
+					return daemonset.Status.DesiredNumberScheduled == daemonset.Status.NumberReady
+				}, metallbutils.DeployTimeout, metallbutils.Interval).Should(BeTrue())
+
+				pods, err = testclient.Client.Pods(OperatorNameSpace).List(context.Background(), metav1.ListOptions{
+					LabelSelector: "component=speaker"})
+				Expect(err).ToNot(HaveOccurred())
+				daemonset, err := testclient.Client.DaemonSets(metallb.Namespace).Get(context.Background(), consts.MetalLBDaemonsetName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(len(pods.Items)).To(Equal(int(daemonset.Status.DesiredNumberScheduled)))
+				var speakerContainerFound bool
+				for _, pod := range pods.Items {
+					Expect(pod.Status.Phase).To(Equal(corev1.PodRunning))
+					for _, container := range pod.Spec.Containers {
+						if container.Name == "speaker" {
+							Expect(container.Resources).NotTo(BeNil())
+							Expect(container.Resources.Limits.Cpu().MilliValue()).To(Equal(int64(100)))
+							speakerContainerFound = true
+						}
+					}
+				}
+				Expect(speakerContainerFound).To((Equal(true)))
+				Expect(daemonset.Spec.Template.Spec.PriorityClassName).To(Equal(priorityClassName))
+				Expect(daemonset.Spec.Template.Annotations["test"]).To(Equal("e2e"))
+				Expect(daemonset.Spec.Template.Spec.Affinity.PodAffinity.RequiredDuringSchedulingIgnoredDuringExecution[0].LabelSelector.MatchLabels["app"]).To(Equal("metallb"))
+
+				metallbutils.Delete(metallb)
```

Comment:
Can you add a test where we change a parameter? Will it work?

Reply:
yes, added a test to cover this.